blog-samples/CSharp/FacebookHandover/Primary/Bots/PrimaryBot.cs

```diff
@@ -0,0 +1,213 @@
+                            await FacebookThreadControlHelper.PassThreadControlAsync(_configuration["FacebookPageToken"], receiver, turnContext.Activity.From.Id, text);
+                            break;
+                        }
+                    }
+
+                    break;
+
+                case OPTION_REQUEST_THREAD_CONTROL:
+                case OPTION_REQUEST_THREAD_CONTROL_NICELY:
+                    // Do nothing because the secondary receiver should react to these instead
+                    break;
+
+                default:
+                    await ShowChoices(turnContext, cancellationToken);
+                    break;
+            }
+        }
+
+        protected override async Task OnConversationUpdateActivityAsync(ITurnContext<IConversationUpdateActivity> turnContext, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("PrimaryBot - Processing a ConversationUpdate Activity.");
+
+            var facebookPayload = (turnContext.Activity.ChannelData as JObject)?.ToObject<FacebookPayload>();
+
+            if (facebookPayload != null)
+            {
+                if (facebookPayload.PassThreadControl != null)
+                {
+                    await turnContext.SendActivityAsync($"Thread control is now passed to {facebookPayload.PassThreadControl.NewOwnerAppId} with the message: {facebookPayload.PassThreadControl.Metadata}");
+                    await ShowChoices(turnContext, cancellationToken);
+                }
+            }
+
+            await base.OnConversationUpdateActivityAsync(turnContext, cancellationToken);
+        }
+        protected override async Task OnEventActivityAsync(ITurnContext<IEventActivity> turnContext, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("PrimaryBot - Processing an Event Activity.");
+
+            // Analyze Facebook payload from EventActivity.Value
+            await ProcessFacebookMessage(turnContext, turnContext.Activity.Value, cancellationToken);
+        }
+
+        private async Task<bool> ProcessFacebookMessage(ITurnContext turnContext, object data, CancellationToken cancellationToken)
+        {
+            return await ProcessStandbyPayload(turnContext, data, cancellationToken)
+                || await ProcessFacebookPayload(turnContext, data, cancellationToken);
+        }
+
+        private async Task<bool> ProcessStandbyPayload(ITurnContext turnContext, object data, CancellationToken cancellationToken)
+        {
+            if (turnContext.Activity.Name?.Equals("standby", StringComparison.InvariantCultureIgnoreCase) == true)
+            {
+                var standbys = (data as JObject)?.ToObject<FacebookStandbys>();
+                if (standbys != null)
+                {
+                    foreach (var standby in standbys.Standbys)
+                    {
+                        await OnFacebookStandby(turnContext, standby, cancellationToken);
+                        return true;
```

Comment:
This seems a little odd, to process just the first Standby. I think we should at least send them all to the OnFacebookStandby.